### PR TITLE
Added ui state listener: activity, fragment and view

### LINKED ISSUE #26: Add UI StateListener
Add Activity, fragment and view that implement state listener.

### LINKED ISSUE #30: Create main UI support classes (Activity, Fragment and View)

### PR DESCRIPTION
**Pull request by flaviokreis**
- [x] Added ui state listener: activity, fragment and view

It closes #26 
It closes #30

### CODE REVIEW

File: app/src/main/java/com/github/raulccabreu/redukt/ui/StateView.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+        registerReducer()
+    }
+
+    override fun onDetachedFromWindow() {
+        unregisterReducer()
+        super.onDetachedFromWindow()
+    }
+
+    override fun setVisibility(visibility: Int) {
```
Comment (raulccabreu) [inline app/src/main/java/com/github/raulccabreu/redukt/ui/StateView.kt]: I don't think 100% sure that is necessary a hide/show register and unregister mechanism at views, if we will add this, maybe at activity and fragment its necessary to implement this behavior too at resume/show with the `changeStateWhenHidden` method (suggest to change the method name to: `listenChangesWhenHidden`).

File: app/src/main/java/com/github/raulccabreu/redukt/ui/StateActivity.kt [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package com.github.raulccabreu.redukt.ui
+
+import android.support.v7.app.AppCompatActivity
+import com.github.raulccabreu.redukt.Redukt
+import com.github.raulccabreu.redukt.states.StateListener
+
+abstract class StateActivity<T> : AppCompatActivity(), StateListener<T> {
```
Comment (raulccabreu) [inline app/src/main/java/com/github/raulccabreu/redukt/ui/StateActivity.kt]: What do you think about ReduktView, ReduktActivity and ReduktFragment or ReactiveView, ReactiveActivity....

**Review by raulccabreu — COMMENTED** [review]
Good! Waiting the answers about my doubts :+1:

File: app/src/main/java/com/github/raulccabreu/redukt/ui/StateView.kt [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        super.setVisibility(visibility)
+        if (visibility == View.VISIBLE && !isRegistered)
+            registerReducer()
+        else if (visibility != View.VISIBLE && !changeStateWhenHidden())
+            unregisterReducer()
+    }
+
+    private fun registerReducer() {
```
Comment (niltonvasques) [inline app/src/main/java/com/github/raulccabreu/redukt/ui/StateView.kt]: Hmm! We aren't register it in a reducer, it is registering itself to hear state changes. Thus I think that a more appropriated name could be, registerStateChanges for example or other related.

**Review by niltonvasques — CHANGES_REQUESTED** [review]
Great work man! Those classes make more easy to bring redux to UI layer.

Would be good to have some tests to make sure that those things are working properly.

**Review by niltonvasques — APPROVED** [review]

**Pull request merged**